app/controllers/proposal_forms_controller.rb

```diff
@@ -1,67 +1,95 @@
 class ProposalFormsController < ApplicationController
-  before_action :set_proposal_form, only: %i[edit update show destroy clone proposal_field]
-
+  before_action :set_proposal_type
+  before_action :set_proposal_form, only: %i[edit update show clone proposal_field]
+  
   def index
-    @proposal_forms = ProposalForm.all
+    @proposal_forms = @proposal_type&.proposal_forms
   end
 
   def new
+    forms = @proposal_type.proposal_forms.where(status: [:active, :draft])
+    forms.update_all(status: :inactive)
     @proposal_form = ProposalForm.new
   end
 
-  def edit; end
+  def edit
+    if @proposal_form.active?
+      @proposal_form.update(status: :inactive)
+      form = @proposal_form.deep_clone include: :proposal_fields
+      form.status = :draft
+      form.save
+      redirect_to edit_proposal_type_proposal_form_path(@proposal_type, form)
+    end
+  end
```

Comment:
We should have a "cancel edit" feature, that resets the proposal status to :active, in case the user changes mind about editing.